src/Android/Xamarin.Android/Samples/Layers/KmlNetworkLinks/KmlNetworkLinks.cs

```diff
@@ -0,0 +1,189 @@
+                _myMap.OperationalLayers.Add(myKmlLayer);
+
+                // Define and extent to zoom to (in the case: Frankfurt Germany area)
+                Envelope myExtent = new Envelope(647571.091263907, 6310478.90141415, 1350638.62905514, 6604034.01050928, SpatialReferences.WebMercator);
+
+                // Zoom to the extent
+                await _myMapView.SetViewpointGeometryAsync(myExtent);
+            }
+            catch (Exception ex)
+            {
+                var builder = new AlertDialog.Builder(this);
+                builder.SetMessage("An error occurred" + ex.ToString()).SetTitle("Sample error").Show();
+            }
+        }
+
+        private void KmlLayer_LoadStatusChanged(object sender, Esri.ArcGISRuntime.LoadStatusEventArgs e)
+        {
+            if (e.Status == Esri.ArcGISRuntime.LoadStatus.Loaded)
+            {
+                // This makes sure that we switch to the UI thread before interacting with any UI components
+                RunOnUiThread(() =>
+                {
+                    // Clear the existing text view
+                    _myLabel.Text = "";
+
+                    // Traverse the file and display the refresh intervals for network links
+                    TraverseNodesUpdateStatus(_myKmlDataset.RootNodes);
+                });
+            }
+        }
+
+        private void TraverseNodesUpdateStatus(IReadOnlyList<KmlNode> nodes)
+        {
+            // Iterate through each node
+            foreach (KmlNode content in nodes)
+            {
+                // Get the type of the node
+                Type nodeType = content.GetType();
+
+                // Update the UI if the content is a Network Link
+                if (nodeType == typeof(KmlNetworkLink))
+                {
+                    // Get a reference to the item as a KmlNetworkLink
+                    KmlNetworkLink netLink = content as KmlNetworkLink;
+
+                    // Get the existing text of the label
+                    string existingLabel = _myLabel.Text as string;
+
+                    // Update the text with information about the layer
+                    existingLabel += String.Format("Name: {0} - Refresh Interval: {1}" +
+                        System.Environment.NewLine, netLink.Name, netLink.RefreshInterval);
+
+                    // Update the label with the new text
+                    _myLabel.Text = existingLabel;
+                }
+
+                // Recur if the node type has children
+                if (nodeType == typeof(KmlNetworkLink))
+                {
+                    // Cast the node to the correct type
+                    KmlNetworkLink myKmlNetworkLink = (KmlNetworkLink)content;
+
+                    // Recur on the children of the node
+                    TraverseNodesUpdateStatus(myKmlNetworkLink.ChildNodes);
+                }
+                else if (nodeType == typeof(KmlContainer))
```

Comment:
Type checks written in this way are repetitive and will fail if we add another KmlContainer implementation later. Why not `if (content is KmlContainer)`?

Reply:
(this applies to all other platforms' code as well)